src/Uplink/Auth/Auth_Url_Builder.php

```diff
@@ -35,10 +35,11 @@ public function __construct(
 	 *
 	 * @param  string  $slug  The product/service slug.
 	 * @param  string  $domain  An optional domain associated with a license key to pass along.
+	 * @param  string  $license  An optional license key to pass along.
 	 *
 	 * @return string
 	 */
-	public function build( string $slug, string $domain = '' ): string {
+	public function build( string $slug, string $domain = '', string $license = '' ): string {
```

Comment:
When we had last chatted, we were going to add `->set_license( string $license )` and have it return `$this`.

And then the `build()` signature doesn't have to get longer and can internally observe the presence/absence of the license.

Reply:
I remember talking through both and thought that we settled on just adding an optional parameter because it was simpler than setting up a fluent system for building the url.